app/assets/javascripts/controllers/cloud_volume/cloud_volume_form_controller.js

```diff
@@ -180,5 +177,19 @@ ManageIQ.angular.app.controller('cloudVolumeFormController', ['$scope', 'miqServ
     }
   };
 
+  var formOptions = function() {
+    API.get("/api/providers?expand=resources&attributes=id,name&filter[]=supports_block_storage?=true").then(function(data) {
+      $scope.storageManagers = data.resources;
+
+      if (storageManagerId) {
+        $scope.cloudVolumeModel.storage_manager_id = parseInt(storageManagerId, 10);
+        $scope.storageManagerChanged($scope.cloudVolumeModel.storage_manager_id);
+      }
+
+      $scope.afterGet = true;
+      miqService.sparkleOff();
+    });
+  };
+
```

Comment:
@gberginc Can you add a `.catch` block here to catch failures?

We typically write `API.get` like this --

```javascript
API.get(url)
  .then(getData)
  .catch(miqService.handleFailure);
```